### PR TITLE
Wajid

### PR DESCRIPTION
**Pull request by Wajid-Fareed**

### CODE REVIEW

File: src/app/blinds/[product]/[subproduct]/page.tsx [hunk 1]
```diff
@@ -1,98 +1,95 @@
...
-  const { data: subCategories, isLoading: subLoading } = useQuery<ICategory[]>({
-    queryKey: ['sub-categories'],
-    queryFn: fetchSubCategories,
-  });
+export async function generateMetadata({ params }: { params: { subproduct: string } }): Promise<Metadata> {
+  const { subproduct } = params;
+  const Cateories = [2];
+  const [products] = await Promise.all([
+    fetchProducts(),
+  ]);
```
Comment (MFaad1) [inline src/app/blinds/[product]/[subproduct]/page.tsx:15]: we don't need promise.all method

File: src/app/blog/[name]/Blog.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  } | null>(null);
+  const [catgoryPageSkeleton, setCatgoryPageSkeleton] = useState(false);
+  const [relatedPosts, setRelatedPosts] = useState<BlogInfo[]>([]);
+  const [filterCategoryPosts, setfilterCategoryPosts] = useState<BlogInfo[]>(
+    [],
+  );
+
+  const { data: blogs, isLoading } = useQuery<BlogInfo[]>({
+    queryKey: ['blogs'],
+    queryFn: fetchBlogs,
```
Comment (MFaad1) [inline src/app/blog/[name]/Blog.tsx:39]: We shouldn't call the api's on client side as we are using parent server component

File: src/app/blog/[name]/Blog.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+  return (
+    <>
+      {isLoading || !catgoryPageSkeleton ? (
+        <PageSkelton />
+      ) : catgoryPage ? (
+        <>
+          <TopHero
+            title={catgoryPage?.name || 'blogs'}
+            //@ts-expect-error
```
Comment (MFaad1) [inline src/app/blog/[name]/Blog.tsx:105]: We should fix it rather than adding expected-error

File: src/app/blog/page.tsx [hunk 4]
```diff
@@ -21,6 +23,44 @@ const fetchBlogs = async (): Promise<BlogInfo[]> => {
   }
 };
 
+
+export async function generateMetadata(): Promise<Metadata> {
```
Comment (MFaad1) [inline src/app/blog/page.tsx:27]: We should add meta data object rather than generateMetadata functions

**Review by MFaad1 — CHANGES_REQUESTED** [review]

File: src/app/blinds/[product]/[subproduct]/page.tsx (hunk 1 above)
Comment (Wajid-Fareed) [reply src/app/blinds/[product]/[subproduct]/page.tsx:15]: fixed

File: src/app/blog/[name]/Blog.tsx (hunk 2 above)
Comment (Wajid-Fareed) [reply src/app/blog/[name]/Blog.tsx:39]: fixed

File: src/app/blog/[name]/Blog.tsx (hunk 3 above)
Comment (Wajid-Fareed) [reply src/app/blog/[name]/Blog.tsx:105]: fixed

File: src/app/blog/page.tsx (hunk 4 above)
Comment (Wajid-Fareed) [reply src/app/blog/page.tsx:27]: fixed

**Review by MFaad1 — APPROVED** [review]

**Pull request merged**